pkg/action/status.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright The Helm Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package action
+
+import (
+	"helm.sh/helm/v3/pkg/action"
+)
+
+// Status is the action for checking the deployment status of releases.
+//
+// It provides the implementation of 'helm status'.
+type Status struct {
+	*action.Status
```

Comment:
I think that as it is, we are using Helm's `Configuration` type inside of the embedded `helm/pkg/action.Status` struct. But we have our own Hypper's `Configuration` type that we should be using. One needs to overload the `cfg` here:

```suggestion
	*action.Status
	 cfg *Configuration
```

If we don't overload, it may fail somewhere. We don't have unit tests on `NewStatus()` (Helm doesn't, either, because it seems to be a simple enough pkg). But at some point it would have failed.

Reply:
As much as we can we should directly use the Helm packages as packages rather than copying code over. This way we inherit updates such as bug and security fixes.

Reply:
For what is worth, our `Configuration` type is also a composite type of Helm's `Configuration` type, exactly for that reason: to import Helm code and not copy it.

Reply:
Dont we do the same on the `List` part? What is the difference?

Also, when calling `NewStatus` arent we passing our `Configuration` found at `hypper/pkg/action/action.go` which is a composite of our+helm one? I agree that using first helm config and then overloading ours is not ideal, as that could lead to bugs.

Like, I can change this, but we should agree on how to deal with this to make it standard across our code.

Reply:
@Itxaka  I think you are right, and actually List should be implemented as:

```
type List struct {
	*action.List
	cfg *Configuration
}
```

Aaaand that was an oversight from me when I implemented it. We should fix that in a follow-up PR :).

Reply:
Also we should not access the underlying helm code directly in a composition, this can and will cause weird problems, if needed overload and access indirectly through our own types.